Multiphysics divertor monoblock

simopier opened the pull request:
(Ref. #54)

Note that the addition of TMAP8 into SALAMANDER is redundant with https://github.com/idaholab/salamander/pull/53 and should be removed before merging. However, this is needed here to develop this example case before  https://github.com/idaholab/salamander/pull/53 gets merged.

grmnptr committed:
Update submodule. (#21)

gsgall committed:
Adding efield tests for boris stepper

gsgall committed:
Apply suggestions from code review

Co-authored-by: Pierre-Clement Simon <pierreclement.simon@gmail.com>

gsgall committed:
Update doc/content/source/userobjects/ParticleStepperBase.md

Co-authored-by: Pierre-Clement Simon <pierreclement.simon@gmail.com>

gsgall committed:
resolving merge issue

gsgall committed:
Update test/tests/userobjects/leapfrog_stepper/tests

Co-authored-by: Pierre-Clement Simon <pierreclement.simon@gmail.com>

gsgall committed:
Adding PBNC conference reference

gsgall committed:
Adding the PNBC reference

gsgall committed:
Merge branch 'devel' of https://github.com/gsgall/fenix into devel

gsgall committed:
fixing the apostrophe's that broke the tests

loganharbour committed:
Merge pull request #22 from grmnptr/nonlinear-var-fix-21

Replace VAR_NONLINEAR with VAR_SOLVER.

simopier committed:
Merge pull request #20 from gsgall/devel

Closes #19: Adding capability for charged particle motion in electric and magnetic fields

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2148399

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2149283

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2150725

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2151751

gsgall committed:
Closes #25 Adding the ability to solve for the electrostatic potential based on particle positions

gsgall committed:
Apply suggestions from code review

Co-authored-by: Pierre-Clement Simon <pierreclement.simon@gmail.com>

gsgall committed:
Removing comment and updating documentation

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2160840

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2162424

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2164693

gsgall committed:
adding code for understanding how to get the pont of a node

gsgall committed:
commit for checkout

gsgall committed:
Making the execute_on parameter of the charge accumulator a private parameter

gsgall committed:
Removing header includes from ChargeAccumulator.h

- needed to add the Ray.h header for the defintion of RayDataIndex

simopier committed:
Merge pull request #26 from gsgall/devel

Closes #25 Adding the ability to solve for the electrostatic potential based on particle positions

gsgall committed:
Sampling causses hanging in parallel for some reason

gsgall committed:
Unable to get the sampler in the user object

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2168235

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2172112

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2174113

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2177588

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2180779

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2181089

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2182644

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2185710

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2186750

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2191179

cticenhour committed:
Add user and theory manual landing pages

Also set all SQA document defaults to error instead of warning

Refs #2

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2192461

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2193629

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2194172

cticenhour committed:
Fill-in empty template fields for RTM, SDD, SRS, VVR

Refs #2

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2195469

cticenhour committed:
Address review suggestion in SRS

Co-authored-by: Pierre-Clement Simon <pierreclement.simon@gmail.com>

cticenhour committed:
Bump coverage to 90 in both SRS and CI

simopier committed:
Merge pull request #31 from cticenhour/sqa-doco

Fill in empty fields in SQA documentation

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2197867

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2204240

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2205390

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2208052

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2208907

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2210172

simopier committed:
Improve README.md

(Ref. #32)

simopier committed:
Update config.yml to update the top menu ad add new documentation pages

simopier committed:
Add list of publications suporting FENIX development

simopier committed:
Add documentation page for example cases

simopier committed:
Add getting started documentation

simopier committed:
Create verification, validation, and example cases page (and remove folder dedicated to example cases only)

simopier committed:
Create main index page for FENIX website

simopier committed:
Add fenix logo on website

simopier committed:
Series of fixes for the documentation/website

simopier committed:
Additional fixes for the documentation/website

simopier committed:
Fix typo in README.md

simopier committed:
Fix issues with chemical_reactions

simopier committed:
Apply suggestions from code review

Co-authored-by: Casey Icenhour <cticenhour@gmail.com>

simopier committed:
Review comment:
- move documentation about utilities
- add logos and collaborator section
- rename file for proceedings using fenix

simopier committed:
Add link to discussion page on github

simopier committed:
Apply suggestions from code review

Co-authored-by: Casey Icenhour <cticenhour@gmail.com>

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2221730

simopier committed:
Update doc/content/index.md

Co-authored-by: Casey Icenhour <cticenhour@gmail.com>

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2223203

simopier committed:
Update doc/config.yml

Co-authored-by: Casey Icenhour <cticenhour@gmail.com>

cticenhour committed:
Merge pull request #33 from simopier/FENIX_website_update

FENIX website update

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2226324

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2228676

gsgall committed:
Adding files to move to INL machine

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2232607

Commit:
very simple 1D initialization seems to be working

gsgall committed:
Convergence study for 1D is done

gsgall committed:
adding another study increasing the number of particles per element for the element study

gsgall committed:
Updating convergence study to use the l2 error calculator in MOOSE

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2235987

gsgall committed:
2D quads initialization done

gsgall committed:
Adding triangle initialization

gsgall committed:
adding again for being safe

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2238729

gsgall committed:
parallel consistent particle initialization 1D is working

gsgall committed:
Adding 1D parallel consistent test

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2241528

gsgall committed:
particle initialization is done in all 3 dimensions

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2244147

gsgall committed:
pyramid working needing to clean it up quite a bit

gsgall committed:
Finished all of the basic element types

gsgall committed:
current working version for all basic mesh types

gsgall committed:
Rebasing for fenix moose resbase

gsgall committed:
All element types are added as tests now

gsgall committed:
removing uneeded test things

gsgall committed:
Added documentation for the initializer object

gsgall committed:
reformatting

gsgall committed:
Adding some more documentation

gsgall committed:
Closes #36

gsgall committed:
Removing double dot for the build

gsgall committed:
Adding stochastic_tools in unit Makefile and doc/config.yml

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2251096

gsgall committed:
Seperating tests for debugging

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2252159

gsgall committed:
Adding some printing for testing

gsgall committed:
Adding output

gsgall committed:
adding more general printing

gsgall committed:
Adding something else

gsgall committed:
Adding a change that shouldn't matter but might

gsgall committed:
Adding a change to the way reference points are created

gsgall committed:
testing working on both

gsgall committed:
Setting mesh_mode=REPLICATED for pyramids and tets

- This was required because element ids for these types of elements are not consistent when using distributed meshing
	- this does not matter in actual use case but does for testing

gsgall committed:
updating sqa documentation for build

gsgall committed:
Adding stochastic_tools for sqa documentation

gsgall committed:
Apply suggestions from code review

Co-authored-by: Pierre-Clement Simon <pierreclement.simon@gmail.com>

gsgall committed:
Apply suggestions from code review

Co-authored-by: Pierre-Clement Simon <pierreclement.simon@gmail.com>

gsgall committed:
Apply suggestions from code review

Co-authored-by: Pierre-Clement Simon <pierreclement.simon@gmail.com>

gsgall committed:
Apply suggestions from code review

Co-authored-by: Pierre-Clement Simon <pierreclement.simon@gmail.com>

gsgall committed:
Cleaning up some suggestions and using \!include for the test cases

gsgall committed:
updating value comparison logic and testing

gsgall committed:
Adding requested c++ note

simopier committed:
Merge pull request #37 from gsgall/initialization

Uniform Particle Initialization

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2257611

gsgall committed:
Bounding Box IC complete

- Uniform element sampling has been refactored for flexibility
- The element uniform sampler has also been simplified interally
- Closes #42

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2261829

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2263661

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2266637

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2268626

gsgall committed:
Updating FENIX to remove legacy behaviour

- closes #40

loganharbour committed:
Merge pull request #41 from gsgall/devel

Updating FENIX to remove legacy behavior

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2274202

gsgall committed:
Finixing bug and removing copy with node range

gsgall committed:
Fixing documentation error

simopier committed:
Add copyright notice

(Close. #44)

simopier committed:
Update header from files in FENIX

simopier committed:
Add script that automatically checks and update headers

Co-authored-by: Casey Icenhour <casey.icenhour@inl.gov>

simopier committed:
Update ASCII art for fenix to be consistent everywhere

simopier committed:
Apply suggestions from code review

Co-authored-by: Casey Icenhour <cticenhour@gmail.com>

simopier committed:
Add link that fenix will be at after being released open source

simopier committed:
Update scripts/fixup_headers.py

Co-authored-by: Casey Icenhour <cticenhour@gmail.com>

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2278464

gsgall committed:
Apply suggestions from code review

Co-authored-by: Pierre-Clement Simon <pierreclement.simon@gmail.com>
Co-authored-by: Logan Harbour <loganhharbour@gmail.com>

gsgall committed:
reworking to address comments

gsgall committed:
reorganizing some files to make the structure better

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2281474

gsgall committed:
closes #46

loganharbour committed:
Merge pull request #47 from gsgall/hot-fix

Remove Deprecated libMesh smart pointer

gsgall committed:
Updating documentation

gsgall committed:
Moving test locations around a little bit for ease of navigation

gsgall committed:
Fixing documentation

gsgall committed:
Updating bounding box documentation

gsgall committed:
Small documentation modification

gsgall committed:
removing file that should not exist

gsgall committed:
Trying to allow warnings for some bounding box tests

gsgall committed:
Adjusting tests so they work on Apptainer

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2284919

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2285943

gsgall committed:
Adding error in the unsupported element error test for bounding box for apptainer build

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2288079

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2288461

simopier committed:
Merge pull request #45 from simopier/FENIX_copyright_header_notice

Fenix copyright notice

gsgall committed:
Apply suggestions from code review

Co-authored-by: Pierre-Clement Simon <pierreclement.simon@gmail.com>

gsgall committed:
Fixing initializer that I broke accidentally

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2292085

loganharbour committed:
Use public website link

refs #48

loganharbour committed:
Remove leftover from stork

loganharbour committed:
Merge pull request #49 from idaholab/48_public_links

Update website link

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2293502

simopier committed:
Merge pull request #50 from gsgall/initialization

New Bounding Box IC

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2297139

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2299776

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2300337

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2306239

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2309675

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2310342

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2312478

cticenhour committed:
Add TMAP8 submodule

Refs #52

cticenhour committed:
Add TMAP8 to FENIX Makefile and FenixApp.C

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2315030

cticenhour committed:
Update TMAP8 submodule

Refs idaholab/TMAP8#152

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2320970

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2323265

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2325809

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2328700

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2330313

simopier committed:
Create example with Divertor monoblock:
- TMAP8 and Heat conduction in 3D

(Ref. #54)

simopier commented:
@FranklinTrevor @meltawila 
Use this PR as base for further developments.

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/2381781/) on 340a5d9 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/fenix/docs/PRs/55/site)

This comment will be updated on new commits.

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2333184

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2334481

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2338047

simopier committed:
Clean up base input file and add file for mechanics

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2339859

Commit:
add neutronics

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2341581

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2342779

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2344218

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2348729

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2349015

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2349694

simopier committed:
Move instructions to contribute to FENIX

(Close. #57)

simopier committed:
Fix failing test by fixing link

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2353421

cticenhour committed:
Merge pull request #58 from simopier/contributing_instructions_2

Move instructions to contribute to FENIX

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2355726

simopier committed:
Add embedded youtube DOE OTT video on FENIX website

(Close. #59)

simopier committed:
Apply suggestions from code review

Co-authored-by: Casey Icenhour <cticenhour@gmail.com>

cticenhour committed:
Merge pull request #60 from simopier/FENIX_DOE_OTT

Commit:
Updating stable moose submodule refs #000

Triggered by CIVET job https://civet.inl.gov/job/2360036

gsgall committed:
Added capability to place particles on the domain when using the TestInitializedPICStudy for consitency

gsgall committed:
Lieberman example working with new formatting

gsgall committed:
Added a test for the negative gradient aux kernel

- also updated a test input file 'simple_potential_solve' that had a mistaken change

gsgall committed:
Extracted residual accumulator base class

- useful for simplifying future accumulators

gsgall committed:
Added an intial version of a post processor for kinetic data

- For testing the csv files need to have rays in the same order even in parallel
- Parallel consistency in ray data csv files also requires that we be able to assign rays indicies that will be parallel consistent

gsgall committed:
Removed TestPICStudyUserDefinedRays to make tests consistent with the use of the initializer object

- Also cleaned up the tests by removing the RayBCs block from all files which do not require it

gsgall committed:
Updated the uniform grid input file to be parallel consistent

gsgall committed:
Added lieberman casea as a test now documentation is needed

gsgall committed:
Adding current progress

gsgall committed:
Added the script and images needed to reproduce the lieberman example results

- the produced images and the images need to produce the figures are in the doc/content/figures folder
- no actual images were added in the test directy symbolic links between the actual files were added in the directory

gsgall committed:
adding doc page

gsgall committed:
trying to get it to recognize the new page

gsgall committed:
Adding a file for the number density accumulator

gsgall committed:
Added documentation for the lieberman benchmark case, still need to format the figures a bit better

gsgall committed:
Correcting equation for the electrostatic potential in the lieberman example case

gsgall committed:
Generalized particle data output capability and adding more documentation.

gsgall committed:
Adding documentation for the UniformGridParticleInitializer

gsgall committed:
Removed the ray id from the Particle data collector that will actually be used by end users since they do not care about particle meta data

Closes #63, #62, and constributes to #23, #27, #61

gsgall committed:
Applying formatting changes

gsgall committed:
removing print statements from files

cticenhour committed:
Add TMAP8 to installation instructions

Refs #52

cticenhour committed:
Add new content to documentation

- Adds TMAP8 content (requires a forthcoming adjustment to TMAP8 documentation to build successfully)
- Tweaks name of FENIX user notes page in light of TMAP8 conflict
- Adds code coverage link to navigation
- Adds module content
- Add SQA dependencies and SQA requirement content for modules and apps

Refs #52

cticenhour committed:
Update TMAP8 submodule

gsgall committed:
updating comparison figure

cticenhour committed:
Fixup unit Makefile and adjust modules within main Makefile

Refs #52

simopier committed:
Merge pull request #65 from cticenhour/add-tmap8

Add TMAP8 submodule

gsgall committed:
Apply suggestions from code review

Co-authored-by: Casey Icenhour <cticenhour@gmail.com>

gsgall committed:
Applying formatting

gsgall committed:
Adding any missed suggetions

gsgall committed:
Apply suggestions from code review

Co-authored-by: Casey Icenhour <cticenhour@gmail.com>

cticenhour committed:
Merge pull request #64 from gsgall/tutorials

Adding Lieberman benchmark case and clean up before the UKAEA training

cticenhour committed:
Fix User Notes link in documentation to refer to the proper file

The current link uses TMAP8's version of the file. 

Refs #52

simopier committed:
Merge pull request #66 from idaholab/fix-user-notes

Fix User Notes link in documentation to refer to the proper file

simopier committed:
remove trailing whitespaces

simopier committed:
Update end of line

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/2381785/) on 340a5d9 wanted to post the following:


## Coverage

Coverage did not change

[Full coverage report](https://mooseframework.inl.gov/fenix/docs/PRs/55/coverage)



This comment will be updated on new commits.

Commit:
Adding the H20 cooling channel and coupling capability to the main input file

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/2538683/), step Sync to remote on 14044a2 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/fenix/docs/PRs/55/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/2538688/), step Generate coverage on 14044a2 wanted to post the following:


## Coverage

Coverage did not change

[Full coverage report](https://mooseframework.inl.gov/fenix/docs/PRs/55/coverage)



This comment will be updated on new commits.

Commit:
update neutronics

Commit:
clang-format

Commit:
correct file name

meltawila commented:
`divertor_monoblock_base.i` is `divertor_monoblock_HC_TMAP8_Base.i` without functions/temp_flux_bc_func & BC/temp_top

lindsayad commented on test/examples/divertor_monoblock/H20_Cooling_Channel.i:
Shouldn't this be H2O instead of H20, 😄
